kedro-datasets/kedro_datasets/matlab/matlab_dataset.py

```diff
@@ -0,0 +1,157 @@
+            self, 
+            filepath: str,
+            save_args: Dict[str, Any]=None,
+            version: Version =None,
+            credentials: Dict[str, Any] =None,
+            fs_args: Dict[str, Any]=None,
+            metadata: Dict[str, Any]=None)->None:
+        """Creates a new instance of MatlabDataSet to load and save data from/to a MATLAB file.
+
+        Args:
+            filepath: Filepath in POSIX format to a Matlab file prefixed with a protocol like `s3://`.
+                If prefix is not provided, `file` protocol (local filesystem) will be used.
+                The prefix should be any protocol supported by ``fsspec``.
+                Note: `http(s)` doesn't support versioning.
+            save_args: .mat options for saving .mat files.
+            version: If specified, should be an instance of
+                ``kedro.io.core.Version``. If its ``load`` attribute is
+                None, the latest version will be loaded. If its ``save``
+                attribute is None, save version will be autogenerated.
+            credentials: Credentials required to get access to the underlying filesystem.
+                E.g. for ``GCSFileSystem`` it should look like `{"token": None}`.
+            fs_args: Extra arguments to pass into underlying filesystem class constructor
+                (e.g. `{"project": "my-project"}` for ``GCSFileSystem``), as well as
+                to pass to the filesystem's `open` method through nested keys
+                `open_args_load` and `open_args_save`.
+                Here you can find all available arguments for `open`:
+                https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.spec.AbstractFileSystem.open
+                All defaults are preserved, except `mode`, which is set to `r` when loading
+                and to `w` when saving.
+            metadata: Any arbitrary metadata.
+                This is ignored by Kedro, but may be consumed by users or external plugins.
+        """
+        _fs_args = deepcopy(fs_args) or {}
+        _fs_open_args_load = _fs_args.pop("open_args_load", {})
+        _fs_open_args_save = _fs_args.pop("open_args_save", {})
+        _credentials = deepcopy(credentials) or {} 
+
+        protocol, path = get_protocol_and_path(filepath, version)
+        self._protocol = protocol
+        if protocol == "file":
+            _fs_args.setdefault("auto_mkdir", True)
+        self._fs = fsspec.filesystem(self._protocol)
+        self.metadata = metadata
+
+        super().__init__(
+            filepath=PurePosixPath(path),
+            version=version,
+            exists_function=self._fs.exists,
+            glob_function=self._fs.glob,
+        )
+        # Handle default save arguements
+        self._save_args = deepcopy(self.DEFAULT_SAVE_ARGS)
+        if save_args is not None:
+            self._save_args.update(save_args)
+        
+        _fs_open_args_save.setdefault("mode"< "w")
```

Comment:
What is this `"mode"<"w"` doing? Maybe it's a typo?

Reply:
That line sets the default mode of saving to 'write'. Yes that is a typo.